pkg/cloudprovider/huaweicloud/security_group_listener.go

```diff
@@ -0,0 +1,130 @@
+package huaweicloud
+
+import (
+	"context"
+
+	ecs "github.com/huaweicloud/huaweicloud-sdk-go-v3/services/ecs/v2"
+	ecsmodel "github.com/huaweicloud/huaweicloud-sdk-go-v3/services/ecs/v2/model"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/klog/v2"
+)
+
+type SecurityGroupListener struct {
+	cloudProvider    *CloudProvider
+	ecsApiClient     *ecs.EcsClient
+	cloudConfigSecId string
+}
+
+func (s *SecurityGroupListener) startSecurityGroupListener() error {
+	watcher, err := s.cloudProvider.kubeClient.Nodes().Watch(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		klog.Errorf("failed to watch kubernetes cluster node list: %s", err)
+		return err
+	}
+
+	for {
+		for event := range watcher.ResultChan() {
+			newNode, ok := event.Object.(*v1.Node)
+			if !ok {
+				klog.Error("unexpected type:%v", event.Type)
+				continue
+			}
+			switch event.Type {
+			case watch.Added:
+				node, err := s.cloudProvider.Basic.ecsClient.GetByNodeName(newNode.Name)
+				if err != nil {
+					klog.Error("Add node: can not get kubernetes node name: %v", err)
+					continue
+				}
+				if node.Name != newNode.Name {
```

Comment:
`node.Name` is the ECS name, `newNode.Name` is the k8s cluster node name, they can be different.